### PR TITLE
[plugin] 0.0.1 of donation plugin and invoice query service

### PR DESCRIPTION
**Pull request by renepickhardt**
I have created a plugin that allows c-lightning nodes to spin up a flask webserver which wraps the invoice RPC command. In this way owners of lightning nodes have a "donation"-web address for spontaneous payments.

I plan to activate this plugin tomorrow on my website and do my next youtube video coming Monday about plugin development for c-lightning. Therefor it would be cool if this plugin and #2139 would be available by than on the c-lightning master branch as I can point to them easily during the video that I am about to create. 

Otherwise I will just point to the PR. 

Other than that feedback is highly appriciated. In particular for the java script (somehow eventstream notifications and flask don't run with me) and security of the plugin. My feeling is that this is not the most secure web application.

### CODE REVIEW

File: contrib/plugins/donations/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Note: the modules in the ccan/ directory have their own licenses, but
```
Comment (ZmnSCPxj) [inline contrib/plugins/donations/LICENSE]: I am uncertain why this particular note is here, when the code is in Python and does not (directly) use ccan, or have a `ccan/` subdir within this dir.

Comment (renepickhardt) [reply contrib/plugins/donations/LICENSE]: Cause I copied the file from rusty and forgot to remove this stament. Will fix this later

Comment (ZmnSCPxj) [reply contrib/plugins/donations/LICENSE]: Ping

File: contrib/plugins/donations/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+A Simple HTTP Server is created that can serve a donation webpage and allow to issue invoices.
+The plugin takes one of the following three commands {start/stop/restart} as the first agument
+By default the plugin starts the server on port 8088. This can however be changed with the
+port argument.
+
+this means after starting `lightningd` together with the plugin you can run: `lightning-cli donationserver start` and access the server at http://localhost:8088/donation (in case you run your lightning node at localhost)
+
+## About the plugin
+You can see a demo of the plugin on the authors website at:
+https://ln.rene-pickhardt.de/donation
```
Comment (conscott) [inline contrib/plugins/donations/README.md]: Looks like this is giving a 500 error right now

**Comment by conscott** [comment]
I was unable to get the donation server running locally. After calling 
```
lightning-cli donationserver start
```
It seems the python instance crashes, but I haven't been able to dig into why just yet

I think once this is rebased off of #2188 it may be easier to debug. Ill keep trying when I have more time!

**Comment by cdecker** [comment]
I added the plugin to the plugins repo, to give it a new home where it can be found and improved.

https://github.com/lightningd/plugins/tree/master/donations

Closing this PR :wink:

**Pull request closed without merging**